### PR TITLE
Move settings aliases to lua (#238)

### PR DESCRIPTION
**Pull request by singularpoint**
* remove commands related to settings
* remove core events of displaying settings
* add lua module to handle lua commands
* add scripting docs related to settings
* add setting update core event to handle
  history_saving atomic value correctly
* add lua aliases /settings and /set

### CODE REVIEW

File: resources/lua/macros.lua [hunk 1]
```diff
@@ -60,3 +60,40 @@ end)
...
+		else
+			value = matches[3] == "on"
+			settings.set(key, value)
+		end
+		local key_format = cformat("<yellow>%s<reset>", key)
+		local value_format
+		if value then
+			value_format = cformat("<bwhite>on<reset>")
+		else
+			value_format = cformat("<bblack>off<reset>")
```
Comment (LiquidityC) [inline resources/lua/macros.lua:95]: Don't use "bblack" for off here. That'll be the background color on some terminals (like mine :grin:). I'd prefer bgreen for on and bred for off.

File: resources/lua/macros.lua [hunk 2]
```diff
@@ -60,3 +60,40 @@ end)
...
+		end
+		blight.output(cformat("%s => %s", key_format, value_format))
```
Comment (LiquidityC) [inline resources/lua/macros.lua:97]: Prefix this string with `[**] ` to conform with the old functionality. Also it's the blightmud standard for printing "info".

```lua
blight.output(cformat("[**] %s => %s", key_format, value_format))
```
This hasn't become the standard for all the internal lua generated output. But I'd like to move in that direction instead.

**Review by LiquidityC — CHANGES_REQUESTED** [review]
Nice work. Just a few comments on UX/Visuals

File: resources/lua/macros.lua (hunk 1 above)
Comment (singularpoint) [reply resources/lua/macros.lua]: Done in the updated commit.

File: resources/lua/macros.lua (hunk 2 above)
Comment (singularpoint) [reply resources/lua/macros.lua]: Done in the updated commit.

**Review by LiquidityC — APPROVED** [review]

**Pull request merged**